runtime/src/system_instruction_processor.rs

```diff
@@ -37,12 +37,6 @@ fn create_system_account(
         return Err(SystemError::AccountAlreadyInUse.into());
     }
 
-    // guard against sysvars being made
-    if sysvar::check_id(&program_id) {
-        debug!("CreateAccount: program id {} invalid", program_id);
-        return Err(SystemError::InvalidProgramId.into());
-    }
-
     if sysvar::is_sysvar_id(&to.unsigned_key()) {
```

Comment:
this one can die, IMHO

Reply:
two different checks, I think. This one prevents someone to store account against any sysvar pubkey, while the other check is to prevent assignment of sysvar program to any normal account.